owasp-top10-2017-apps/a10/games-irados/app/routes.py

```diff
@@ -70,78 +73,128 @@ def decorated_function(*args, **kwargs):
+        log_content["outcome"] = "login successful"
+        logger.info(log_content)
         return redirect('/home')
     else:
         return render_template('login.html')
 
+
 @app.route('/register', methods=['GET', 'POST'])
 def newuser():
     if request.method == 'POST':
+        log_content = {
+            "ip_adress": request.remote_addr,
+            "username": "unknown",
+            "message": "",
+            "request_route": request.path,
+            "request_method": request.method,
+            "response": "200",
+            "outcome": ""
+        }
         username = request.form.get('username').encode('utf-8')
         psw1 = request.form.get('password1').encode('utf-8')
         psw2 = request.form.get('password2').encode('utf-8')
-
+        log_content["message"] = "Register attempt"
         if psw1 == psw2:
             psw = Password(psw1)
             hashed_psw = psw.get_hashed_password()
             message, success = database.insert_user(username, hashed_psw)
             if success == 1:
-                logger.info("User added")
+                log_content["outcome"] = "Register successful"
+                logger.info(log_content)
                 flash("Novo usuario adicionado!", "primary")
                 return redirect('/login')
             else:
-                logger.error(message)
+                log_content["status"] = "500"
+                log_content["outcome"] = "Register failed"
+                logger.error(log_content)
                 flash(message, "danger")
                 return redirect('/register')
-        
-        logger.error("Passwords don't match")
+        log_content["outcome"] = "Passwords don't match"
+        log_content["status"] = "500"
+        logger.error(log_content)
         flash("Passwords must be the same!", "danger")
         return redirect('/register')
     else:
         return render_template('register.html')
 
+
 @app.route('/home', methods=['GET'])
 @login_required
 def home():
     return render_template('index.html')
 
+
 @app.route('/coupon', methods=['GET', 'POST'])
 @login_required
 def cupom():
     if request.method == 'POST':
+        log_content = {
+            "ip_adress": request.remote_addr,
+            "username": session.get('username'),
+            "message": "",
+            "request_route": request.path,
+            "request_method": request.method,
+            "response": "200",
+            "outcome": ""
+        }
+        log_content["message"] = "Coupon retrieval attempt"
         coupon = request.form.get('coupon')
-        rows, success = database.get_game_coupon(coupon, session.get('username'))
+        rows, success = database.get_game_coupon(
+            coupon, session.get('username'))
         if not success or rows == None or rows == 0:
-            logger.error("Invalid coupon, no coupon found")
+            log_content["outcome"] = "Invalid coupon"
+            log_content["status"] = "422"
```

Comment:
Although this may cause coupon enumeration (you can mitigate this with other techniques like CAPTCHA, rate limite, etc), you can also return a `404` Not found. What do you think?